File: src/BaseHmiTypes/Screens/Controls/HmiAlarmConditionPresentation.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiAlarmConditionPresentation
{
    public HmiAlarmCondition Condition { get; set; }
    public HmiProperty<bool>? Display { get; set; }
    public HmiProperty<bool>? Blink { get; set; }
    public HmiProperty<bool>? UseAlarmColors { get; set; }
    public HmiProperty<HmiColor>? BackgroundColor { get; set; }
    public HmiProperty<HmiColor>? ForegroundColor { get; set; }
}

File: typescript/base-hmi-types/src/screens/controls/HmiAlarmConditionPresentation.ts
import { HmiColor, HmiProperty } from "../base.js";
import { HmiAlarmCondition } from "./HmiAlarmCondition.js";

export class HmiAlarmConditionPresentation {
  condition = HmiAlarmCondition.ActiveAcknowledged;
  display?: HmiProperty<boolean>;
  blink?: HmiProperty<boolean>;
  useAlarmColors?: HmiProperty<boolean>;
  backgroundColor?: HmiProperty<HmiColor>;
  foregroundColor?: HmiProperty<HmiColor>;
}
